packages/@aws-cdk/aws-appsync/README.md

```diff
@@ -13,186 +13,148 @@
 ---
 <!--END STABILITY BANNER-->
 
-This module is part of the [AWS Cloud Development Kit](https://github.com/aws/aws-cdk) project.
+The `@aws-cdk/aws-appsync` package contains constructs for building flexible
+APIs that use GraphQL. 
 
-## Usage Example
+### Example
 
-Given the following GraphQL schema file `schema.graphql`:
+GraphQL schema file `schema.graphql`:
 
-```graphql
-type ServiceVersion {
-    version: String!
+```gql
+type demo {
+  id: String!
+  version: String!
 }
-
-type Customer {
-    id: String!
-    name: String!
+type Query {
+  getDemos: [ test! ]
 }
-
-input SaveCustomerInput {
-    name: String!
+input DemoInput {
+  version: String!
 }
-
-type Order {
-    customer: String!
-    order: String!
+type Mutation {
+  addDemo(input: DemoInput!): demo
 }
+```
 
-type Query {
-    getServiceVersion: ServiceVersion
-    getCustomers: [Customer]
-    getCustomer(id: String): Customer
-}
+Stack file `app-stack.ts` with auth `IAM`:
 
-input FirstOrderInput {
-    product: String!
-    quantity: Int!
-}
+```ts
+import * as appsync from '@aws-cdk/aws-appsync';
+import * as db from '@aws-cdk/aws-dynamodb';
+
+const api = new appsync.GraphQLApi(stack, 'Api', {
+  name: 'demo',
+  schemaDefinitionFile: join(__dirname, 'schema.graphql'),
+  authorizationConfig: {
+    defaultAuthorization: {
+      authorizationType: appsync.AuthorizationType.IAM
+    },
+  },
+});
+
+const demoTable = new db.Table(stack, 'DemoTable', {
+  partitionKey: {
+    name: 'id',
+    type: AttributeType.STRING,
+  },
+});
+
+const demoDS = api.addDynamoDbDataSource('demoDataSource', 'Table for Demos"', demoTable);
+
+demoDS.createResolver({
+  typeName: 'Query',
+  fieldName: 'getDemos',
+  requestMappingTemplate: MappingTemplate.dynamoDbScanTable(),
+  responseMappingTemplate: MappingTemplate.dynamoDbResultList(),
+});
+
+demoDS.createResolver({
+  typeName: 'Mutation',
+  fieldName: 'addDemo',
+  requestMappingTemplate: MappingTemplate.dynamoDbPutItem(PrimaryKey.partition('id').auto(), Values.projecting('demo')),
+  responseMappingTemplate: MappingTemplate.dynamoDbResultItem(),
+});
```

Comment:
might  be worth adding comments to describe what this is doing as it's not likely to be intuitive to a new user